scripts/generate-api-protos.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# https://github.com/protocolbuffers/protobuf/releases
+readonly PROTOC_RELEASE=3.11.4
+PROTO_ZIP_RELEASE_MD5_LINUX=7c0babfc7d2ae4eff6ce3e47c2de90c2
+PROTO_ZIP_RELEASE_MD5_OSX=58c8716eabdbc1259d14880ace6e719a
+
+# Infer paths.
+readonly REPO_ROOT="$(realpath "$(dirname "${BASH_SOURCE[0]}")/..")"
+readonly API_ROOT="${REPO_ROOT}/api"
+
+# Isolate `go install` commands to the repository. We'll also use this path for other binaries.
+export GOBIN="${REPO_ROOT}/bin"
+
+# Get the directory that the go module is stored in.
+modpath() {
+  set -e
+
+  # Ensure we have the correct version of the module installed.
+  go mod download "${1}"
+
+  go list -f "{{ .Dir }}" -m "${1}"
+}
+
+# Excecute protoc with all the trimmings.
+protoc_command() {
+  echo "${1}"
+  "${PROTOC_BIN}" "${IPATHS[@]}" \
+    --go_out=paths=source_relative:./pkg \
+    --plugin=protoc-gen-validate="${GOBIN}/protoc-gen-validate" \
+    --validate_out="lang=go:pkg/" \
+    --plugin=protoc-gen-go="${GOBIN}/protoc-gen-go" \
+    "${1}"
+}
+
+main() {
+  # Install required plugins for generation. These deps are pinned in go.mod.
+  go install \
+    github.com/envoyproxy/protoc-gen-validate \
```

Comment:
can we also put a go version check to ensure version 1.14? I was trying this out and figured `make compile-protos` and `make compile` were giving me messages that were hard to comprehend.

Reply:
yeah, parsing might break between verrsions..but may be a check for 1.14 in the response

Reply:
I feel this is too brittle. How about we handle this in the documentation? Maybe make it very explicit in the main `README`?

Reply:
sounds good.